src/components/CreateNimi/partials/NimiPreviewCard/NimiPreviewCard.tsx

```diff
@@ -23,19 +23,46 @@ const PreviewFrame = styled(Frame)`
 export function NimiPreviewCard({ nimi }: NimiPreviewCardProps) {
   const [previewNimi, setPreviewNimi] = useState<Nimi>();
 
+  const removeInvalidLinks = async (links) => {
+    const result = await Promise.all(
+      links.map((link) =>
+        nimiLinkValidator
+          .isValid({
+            type: link.type,
+            content: link.content,
+          })
+          .then((isLinkValid) => {
+            console.log('isLinkValidFilter', isLinkValid);
+            return isLinkValid;
+          })
+          .catch((error) => {
+            console.log('error', error);
+            return false;
+          })
+      )
+    );
+
+    return links.filter((_, index) => result[index]);
+  };
```

Comment:
Reduce is also an option.

Reply:
what do you mean? This is temporary function which filters out invalid links so that card doesn't throw out error....We will fix this in the future with sanitisation as we said but for now this is best option...

Reply:
He means I could've used `Array.prototype.reduce()`, as well. I agree Adam. 😄

P.S. Those `console.log`s need to be removed.